charms/istio-gateway/tests/unit/test_charm.py

```diff
@@ -111,3 +111,30 @@ def test_removal(configured_harness, kind, mocked_client, mocker):
     mocked_client.return_value.delete.side_effect = api_error
     with pytest.raises(ApiError):
         configured_harness.charm.on.remove.emit()
+
+
+def test_service_type_cluserip(
+    configured_harness_only_ingress, gateway_service_type, mocked_client
```

Comment:
Will need docstring, otherwise linting will fail.

Reply:
This is not tested in the branch and has been already updated in more recent code.